scala-generator/src/main/scala/models/http4s/ScalaGeneratorUtil.scala

```diff
@@ -37,18 +37,19 @@ class ScalaGeneratorUtil(config: ScalaClientMethodConfig) extends scala.generato
         encodeValue(varName, body.datatype)
       }
 
-      Some(s"val payload = $payload")
+      Some(s"val (payload, formPayload) = (Some($payload), None)")
 
     } else {
       val params = op.formParameters.map { param =>
         val value = encodeValue(param.asScalaVal, param.datatype)
-        s""""${param.originalName}" -> ${value}.asJson"""
+        s""""${param.originalName}" -> ${value}.asJson.noSpaces"""
       }.mkString(",\n")
       Some(
         Seq(
-          "val payload = Map(",
+          "val payload = None",
+          "val formPayload = Some { import io.circe.syntax._ ; org.http4s.UrlForm(",
```

Comment:
Can you move importing io.circe.syntax._ away from here?

Reply:
Here would be better https://github.com/apicollective/apibuilder-generator/blob/master/scala-generator/src/main/scala/models/http4s/Http4sClient.scala#L26

Reply:
@amarrella done